### PR TITLE
MF-1718 - Use static code analysis in CI

### LINKED ISSUE #1718: Use static code analysis in CI
**ENHANCEMENT**
1. Describe the enhancement you are requesting. Enhancements include:
   - tests
   - code refactor
      
     GolangCI linter recommends lot of things like `Error return value not checked`, `unused type`, `deprecates ` and `no value of type uint64 is less than 0`
 To view all the CI check run the command `golangci-lint run `
   - documentation
   - research
   - tooling

2. Indicate the importance of this enhancement to you (must-have, should-have, nice-to-have).
    should-have / nice-to-have

### PR DESCRIPTION
**Pull request by AryanGodara**
### What does this do?
This PR makes the golangci-lint test pass.
### Which issue(s) does this PR fix/relate to?
Resolves #1718 

### List any changes that modify/break current functionality
Behaviour of some functions has changed, and has to be discussed before finalising
### Have you included tests for your changes?
No new tests
### Did you document any new/modified functionality?
N/A
### Notes
N/A

### CODE REVIEW

File: auth/api/grpc/setup_test.go
Comment (AryanGodara) [inline auth/api/grpc/setup_test.go:1]: I need a review of this setup test for starting grpc server

File: things/api/auth/grpc/setup_test.go
Comment (AryanGodara) [inline things/api/auth/grpc/setup_test.go:1]: I also need a review of this grpc server code

File: auth/api/grpc/endpoint.go [hunk 1]
```diff
@@ -35,6 +36,9 @@ func issueEndpoint(svc auth.Service) endpoint.Endpoint {
 }
 
 func identifyEndpoint(svc auth.Service) endpoint.Endpoint {
+	fmt.Println()
```
Comment (rodneyosodo) [inline auth/api/grpc/endpoint.go]: Remove debug statements

File: tools/mqtt-bench/results.go [hunk 2]
```diff
@@ -157,7 +157,10 @@ func printResults(results []*runResults, totals *totalResults, format string, qu
 			log.Printf("Failed to prepare results for printing - %s\n", err.Error())
 		}
 		var out bytes.Buffer
-		json.Indent(&out, data, "", "\t")
+		err = json.Indent(&out, data, "", "\t")
```
Comment (rodneyosodo) [inline tools/mqtt-bench/results.go]: Inline error check

File: tools/mqtt-bench/bench.go [hunk 3]
```diff
@@ -110,7 +116,10 @@ func getBytePayload(size int, m message) (handler, error) {
 		sz := size - n
 		for {
 			b = make([]byte, sz)
-			rand.Read(b)
+			_, err = rand.Read(b)
```
Comment (rodneyosodo) [inline tools/mqtt-bench/bench.go]: Inline error checking

File: tools/mqtt-bench/bench.go [hunk 4]
```diff
@@ -26,7 +26,13 @@ func Benchmark(cfg Config) {
 	var caByte []byte
 	if cfg.MQTT.TLS.MTLS {
 		caFile, err := os.Open(cfg.MQTT.TLS.CA)
-		defer caFile.Close()
+
+		defer func() {
+			err = caFile.Close()
```
Comment (rodneyosodo) [inline tools/mqtt-bench/bench.go]: Inline error checking

File: tools/mqtt-bench/bench.go [hunk 5]
```diff
@@ -26,7 +26,13 @@ func Benchmark(cfg Config) {
...
+			if err != nil {
+				fmt.Println(err)
```
Comment (rodneyosodo) [inline tools/mqtt-bench/bench.go]: Remove debug statements

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by dborovcanin — CHANGES_REQUESTED** [review]
Please update the linter version in CI script and extract it to the variable (like `GOLANGCI_LINT_VERSION=v1.52.2`). Also, please check the linter config [here](https://golangci-lint.run/usage/quick-start/) and find the best combination of linters for us.

**Comment by AryanGodara** [comment]
> Please update the linter version in CI script and extract it to the variable (like `GOLANGCI_LINT_VERSION=v1.52.2`). Also, please check the linter config [here](https://golangci-lint.run/usage/quick-start/) and find the best combination of linters for us.

Removing `deadcode` as it's abandoned by the user, and is replaced by `unused`
Adding errcheck to make sure no error is ignored (unless by choice)
goconst can be useful -> Find repeated strings that can be replaced with constant

These two can be useful :-
unconvert -> Reports unnecessary type conversions
unparam -> Reports unused function parameters
ineffassign -> check ineffectual assignments (May cause lots of failed CIs as this is very strict checker)

File: auth/api/grpc/setup_test.go [hunk 6]
```diff
@@ -4,15 +4,86 @@
...
+
+	go func(done <-chan bool) {
+		for {
+			select {
+			case serverErr <- server.Serve(listener):
+				close(serverErr)
+				return
+			case <-done:
+				close(serverErr)
+				// serverErr <- nil
```
Comment (dborovcanin) [inline auth/api/grpc/setup_test.go]: Remove commented code.

File: auth/api/grpc/setup_test.go [hunk 7]
```diff
@@ -4,15 +4,86 @@
...
+	testRes := make(chan int)
+
+	svc = newService(t)
+	startGRPCServer(t, serverErr, svc, port)
+
+	go func() {
+		for {
+			select {
+			case code := <-testRes:
+				os.Exit(code)
```
Comment (dborovcanin) [inline auth/api/grpc/setup_test.go]: This case should (try to) gracefully stop the gRPC server and release OS resources, not only exit.

File: auth/api/grpc/setup_test.go [hunk 8]
```diff
@@ -4,15 +4,86 @@
...
+func startGRPCServer(t *testing.T, serverErr chan error, svc auth.Service, port int) {
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+	require.Nil(t, err, fmt.Sprintf("got unexpected error while creating new listerner: %s", err))
+
+	server := grpc.NewServer()
+	mainflux.RegisterAuthServiceServer(server, grpcapi.NewServer(mocktracer.New(), svc))
+
+	done := make(chan bool)
+
+	t.Cleanup(func() {
```
Comment (dborovcanin) [inline auth/api/grpc/setup_test.go]: This will never get called in case of successful test execution (due to `os.Exit`) which means that this server will never terminate gracefully.

File: auth/api/grpc/setup_test.go [hunk 9]
```diff
@@ -4,15 +4,86 @@
...
+		close(done)
+	})
+
+	go func(done <-chan bool) {
+		for {
+			select {
+			case serverErr <- server.Serve(listener):
```
Comment (dborovcanin) [inline auth/api/grpc/setup_test.go]: Check how to stop the server correctly (by calling `server.Stop()`).

File: auth/postgres/groups.go [hunk 10]
```diff
@@ -479,7 +482,11 @@ func (gr groupRepository) Unassign(ctx context.Context, groupID string, ids ...s
 		}
 
 		if _, err := tx.NamedExecContext(ctx, qDel, dbg); err != nil {
-			tx.Rollback()
+			err = tx.Rollback()
```
Comment (dborovcanin) [inline auth/postgres/groups.go]: This err shadows the previous one. Please wrap instead.

File: readers/mongodb/messages.go [hunk 11]
```diff
@@ -101,7 +101,10 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) bson.D {
 	if err != nil {
 		return filter
 	}
-	json.Unmarshal(meta, &query)
+	err = json.Unmarshal(meta, &query)
+	if err != nil {
+		return filter
+	}
```
Comment (dborovcanin) [inline readers/mongodb/messages.go]: Inline `if err := json.Unmarshal(....`. What if err is nil?

File: readers/timescale/messages.go [hunk 12]
```diff
@@ -121,7 +121,10 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) string {
 	if err != nil {
 		return condition
 	}
-	json.Unmarshal(meta, &query)
+	err = json.Unmarshal(meta, &query)
+	if err != nil {
+		return condition
+	}
```
Comment (dborovcanin) [inline readers/timescale/messages.go]: Same.

File: things/postgres/things.go [hunk 13]
```diff
@@ -48,7 +48,9 @@ func (tr thingRepository) Save(ctx context.Context, ths ...things.Thing) ([]thin
 		}
 
 		if _, err := tx.NamedExecContext(ctx, q, dbth); err != nil {
-			tx.Rollback()
+			if err := tx.Rollback(); err != nil {
```
Comment (dborovcanin) [inline things/postgres/things.go]: The same remark for error shadowing.

File: tools/mqtt-bench/bench.go [hunk 14]
```diff
@@ -26,9 +26,14 @@ func Benchmark(cfg Config) {
...
-		defer caFile.Close()
+
+		defer func() {
+			if err = caFile.Close(); err != nil {
+				log.Printf("Could  not close file: %s", err)
+			}
+		}()
 		if err != nil {
-			fmt.Println(err)
+			log.Print(err)
```
Comment (dborovcanin) [inline tools/mqtt-bench/bench.go]: Why is this changed?

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: tools/mqtt-bench/bench.go (hunk 14 above)
Comment (AryanGodara) [reply tools/mqtt-bench/bench.go]: Rodney left a comment saying this was a debug statement. But this has always been a print statement. So I changed it to log (so that it will give more infrormation when printed)

Comment (dborovcanin) [reply tools/mqtt-bench/bench.go]: In that case, we should use our logger here.

File: readers/mongodb/messages.go (hunk 11 above)
Comment (AryanGodara) [reply readers/mongodb/messages.go]: I inlined the err.
And if the err is nil, the query is ranged over in a for loop.

File: lora/mqtt/sub.go [hunk 15]
```diff
@@ -54,5 +54,8 @@ func (b broker) handleMsg(c mqtt.Client, msg mqtt.Message) {
 		return
 	}
 
-	b.svc.Publish(context.Background(), &m)
+	err := b.svc.Publish(context.Background(), &m)
+	if err != nil {
```
Comment (rodneyosodo) [inline lora/mqtt/sub.go]: inline error checking

File: opcua/gopcua/subscribe.go [hunk 16]
```diff
@@ -111,7 +111,12 @@ func (c client) Subscribe(ctx context.Context, cfg opcua.Config) error {
 	if err != nil {
 		return errors.Wrap(errFailedSub, err)
 	}
-	defer sub.Cancel()
+	defer func() {
+		err = sub.Cancel()
+		if err != nil {
```
Comment (rodneyosodo) [inline opcua/gopcua/subscribe.go]: inline

File: readers/cassandra/messages.go [hunk 17]
```diff
@@ -128,7 +128,10 @@ func buildQuery(chanID string, rpm readers.PageMetadata) (string, []interface{})
 	if err != nil {
 		return condCQL, vals
 	}
-	json.Unmarshal(meta, &query)
+	err = json.Unmarshal(meta, &query)
```
Comment (rodneyosodo) [inline readers/cassandra/messages.go]: inline

File: readers/mocks/messages.go [hunk 18]
```diff
@@ -39,8 +39,14 @@ func (repo *messageRepositoryMock) ReadAll(chanID string, rpm readers.PageMetada
 	}
 
 	var query map[string]interface{}
-	meta, _ := json.Marshal(rpm)
-	json.Unmarshal(meta, &query)
+	meta, err := json.Marshal(rpm)
+	if err != nil {
+		return readers.MessagesPage{}, err
+	}
+	err = json.Unmarshal(meta, &query)
```
Comment (rodneyosodo) [inline readers/mocks/messages.go]: inline

File: readers/postgres/messages.go [hunk 19]
```diff
@@ -123,7 +123,10 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) string {
 	if err != nil {
 		return condition
 	}
-	json.Unmarshal(meta, &query)
+	err = json.Unmarshal(meta, &query)
```
Comment (rodneyosodo) [inline readers/postgres/messages.go]: inline

File: things/redis/streams.go [hunk 20]
```diff
@@ -49,7 +49,11 @@ func (es eventStore) CreateThings(ctx context.Context, token string, ths ...thin
 			MaxLenApprox: streamLen,
 			Values:       event.Encode(),
 		}
-		es.client.XAdd(ctx, record).Err()
+
+		err = es.client.XAdd(ctx, record).Err()
```
Comment (rodneyosodo) [inline things/redis/streams.go]: inline

File: things/redis/streams.go [hunk 21]
```diff
@@ -134,7 +136,10 @@ func (es eventStore) CreateChannels(ctx context.Context, token string, channels
 			MaxLenApprox: streamLen,
 			Values:       event.Encode(),
 		}
-		es.client.XAdd(ctx, record).Err()
+		err = es.client.XAdd(ctx, record).Err()
```
Comment (rodneyosodo) [inline things/redis/streams.go]: inline

File: things/redis/streams.go [hunk 22]
```diff
@@ -206,7 +207,10 @@ func (es eventStore) Connect(ctx context.Context, token string, chIDs, thIDs []s
 				MaxLenApprox: streamLen,
 				Values:       event.Encode(),
 			}
-			es.client.XAdd(ctx, record).Err()
+			err := es.client.XAdd(ctx, record).Err()
```
Comment (rodneyosodo) [inline things/redis/streams.go]: inline

File: things/redis/streams.go [hunk 23]
```diff
@@ -229,7 +233,10 @@ func (es eventStore) Disconnect(ctx context.Context, token string, chIDs, thIDs
...
```
Comment (rodneyosodo) [inline things/redis/streams.go]: inline

File: tools/mqtt-bench/bench.go [hunk 24]
```diff
@@ -14,33 +14,42 @@ import (
 	"strconv"
 	"time"
 
+	mainflux_log "github.com/mainflux/mainflux/logger"
```
Comment (rodneyosodo) [inline tools/mqtt-bench/bench.go]:
```suggestion
	mflog "github.com/mainflux/mainflux/logger"
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: auth/api/grpc/setup_test.go [hunk 25]
```diff
@@ -4,15 +4,82 @@
...
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+	require.Nil(t, err, fmt.Sprintf("got unexpected error while creating new listerner: %s", err))
+
+	server := grpc.NewServer()
+	mainflux.RegisterAuthServiceServer(server, grpcapi.NewServer(mocktracer.New(), svc))
+
+	go func(done chan int, server *grpc.Server) {
+		for {
+			select {
+			case serverErr <- server.Serve(listener):
```
Comment (dborovcanin) [inline auth/api/grpc/setup_test.go]: Adding a blocking call to the case is not a good idea.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: auth/api/grpc/setup_test.go [hunk 26]
```diff
@@ -4,15 +4,84 @@
...
+			select {
+			case <-testRes:
+				return
+			case err := <-serverErr:
+				if err != nil {
+					log.Fatalf("gPRC Server Terminated")
+				}
+			}
+		}
+	}()
```
Comment (dborovcanin) [inline auth/api/grpc/setup_test.go]: Do not use `t` outside of  `m.Run`, i.e. outside of the runner. This is way too complex. Simplify to something like:
```go
func TestMain(m *testing.M) {
	serverErr := make(chan error)

	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
	if err != nil {
		// Log end exit.
	}

	svc = newService()
	server := grpc.NewServer()
	mainflux.RegisterAuthServiceServer(server, grpcapi.NewServer(mocktracer.New(), svc))

	// Start gRPC server in detached mode.
	go func() {
		serverErr <- server.Serve(listener)
	}()

	code := m.Run()

	server.GracefulStop()
	err = <-serverErr
	if err != nil {
		// Log end exit with 1.
	}
	close(serverErr)
	os.Exit(code)
}

func newService() auth.Service {
	repo := mocks.NewKeyRepository()
	groupRepo := mocks.NewGroupRepository()
	idProvider := uuid.NewMock()

	mockAuthzDB := map[string][]mocks.MockSubjectSet{}
	mockAuthzDB[id] = append(mockAuthzDB[id], mocks.MockSubjectSet{Object: authoritiesObj, Relation: memberRelation})
	ketoMock := mocks.NewKetoMock(mockAuthzDB)

	tokenizer := jwt.New(secret)

	return auth.New(repo, groupRepo, idProvider, tokenizer, ketoMock, loginDuration)
}
```

File: cmd/certs/main.go [hunk 27]
```diff
@@ -169,13 +169,14 @@ func loadCertificates(conf config, logger mflog.Logger) (tls.Certificate, *x509.
 	}
 
 	block, _ := pem.Decode(b)
-	if block == nil {
+	switch block {
```
Comment (dborovcanin) [inline cmd/certs/main.go]: Using `if` in this case is fine, preferable even. What was linter's remark?

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: cmd/certs/main.go (hunk 27 above)
Comment (AryanGodara) [reply cmd/certs/main.go]: It was not happy with the `block == nil {}`, so I had to change the format to reverse the check to `block != nil`

File: tools/mqtt-bench/bench.go [hunk 28]
```diff
@@ -14,33 +14,42 @@ import (
 	"strconv"
 	"time"
 
+	mf_log "github.com/mainflux/mainflux/logger"
```
Comment (drasko) [inline tools/mqtt-bench/bench.go]: @dborovcanin do we typically use underscores in import names?

Comment (AryanGodara) [reply tools/mqtt-bench/bench.go]: We have used mflog at other places, I should've checked this 😅

Comment (AryanGodara) [reply tools/mqtt-bench/bench.go]: I've updated the PR, so it's also ready for review @drasko @dborovcanin

Comment (dborovcanin) [reply tools/mqtt-bench/bench.go]: We don't, good catch.

File: twins/api/http/endpoint_twins_test.go [hunk 29]
```diff
@@ -392,7 +392,7 @@ func TestViewTwin(t *testing.T) {
 
 		var resData twinRes
 		err = json.NewDecoder(res.Body).Decode(&resData)
-		assert.Nil(t, err, fmt.Sprintf("%s: got unexpected error while decoding json: %s", tc.desc, err))
+		assert.Nil(t, err, fmt.Sprintf("%s: gotunexpected error while decoding response body: %s\n", tc.desc, err))
```
Comment (dborovcanin) [inline twins/api/http/endpoint_twins_test.go]: Fix typo.

File: bootstrap/api/endpoint_test.go [hunk 30]
```diff
@@ -1157,7 +1159,9 @@ func TestBootstrap(t *testing.T) {
 		assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
 		if tc.secure && tc.status == http.StatusOK {
 			body, err = dec(body)
-			assert.Nil(t, err, fmt.Sprintf("%sGot unexpected error: %s\n", tc.desc, err))
+			if err != nil {
+				assert.Nil(t, err, fmt.Sprintf("%s: unexpected error while decoding body: %s", tc.desc, err))
```
Comment (dborovcanin) [inline bootstrap/api/endpoint_test.go]: This assertion makes no sense - it's we know the error is not nil here.

File: bootstrap/redis/producer/streams_test.go [hunk 31]
```diff
@@ -447,8 +460,8 @@ func TestBootstrap(t *testing.T) {
 
 	lastID := "0"
 	for _, tc := range cases {
-		_, err := svc.Bootstrap(context.Background(), tc.externalKey, tc.externalID, false)
-		assert.True(t, errors.Contains(err, tc.err), fmt.Sprintf("%s: expected %s got %s\n", tc.desc, tc.err, err))
+		_, errr := svc.Bootstrap(context.Background(), tc.externalKey, tc.externalID, false)
```
Comment (dborovcanin) [inline bootstrap/redis/producer/streams_test.go]: Is this `errr` necessary?

File: bootstrap/service.go [hunk 32]
```diff
@@ -273,17 +273,16 @@ func (bs bootstrapService) Remove(ctx context.Context, token, id string) error {
 func (bs bootstrapService) Bootstrap(ctx context.Context, externalKey, externalID string, secure bool) (Config, error) {
 	cfg, err := bs.configs.RetrieveByExternalID(externalID)
 	if err != nil {
-		return cfg, errors.Wrap(ErrBootstrap, err)
+		err = errors.Wrap(ErrBootstrap, err)
```
Comment (dborovcanin) [inline bootstrap/service.go]: When I run the linter, I do not see remarks for the original code here. Please check. Also, feel free to return `Config{}` instead of `cfg`.

File: scripts/ci.sh [hunk 33]
```diff
@@ -4,6 +4,7 @@ GO_VERSION=1.19.4
 PROTOC_VERSION=21.12
 PROTOC_GEN_VERSION=v1.28.1
 PROTOC_GRPC_VERSION=v1.2.0
+GOLANGCI_LINT_VERSION=v1.51.1
```
Comment (dborovcanin) [inline scripts/ci.sh]: Use the latest version.

File: auth/service_test.go [hunk 34]
```diff
@@ -1082,7 +1083,7 @@ func TestAddPolicies(t *testing.T) {
 	assert.Nil(t, err, fmt.Sprintf("got unexpected error: %s", err))
 
 	tmpID := "tmpid"
-	readPolicy := "read"
+	readPolicy := read
```
Comment (dborovcanin) [inline auth/service_test.go:1086]: Same here, the latest version of linter does not recognize this on my local device.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: auth/service_test.go (hunk 34 above)
Comment (AryanGodara) [reply auth/service_test.go:1086]: I just removed this and I'm getting the error of `string 'read' has 3 occurrences, make it a constant` from goconst

File: auth/jwt/tokenizer.go [hunk 35]
```diff
@@ -83,18 +84,21 @@ func (svc tokenizer) Parse(token string) (auth.Key, error) {
...
-		IssuedAt: time.Unix(c.IssuedAt, 0).UTC(),
+		IssuedAt: c.IssuedAt.Time.UTC(),
 	}
-	if c.ExpiresAt != 0 {
-		key.ExpiresAt = time.Unix(c.ExpiresAt, 0).UTC()
+
+	if c.ExpiresAt != nil && c.ExpiresAt.Time.UTC().Unix() != 0 {
+		key.ExpiresAt = c.ExpiresAt.Time.UTC()
+	} else {
+		key.ExpiresAt = time.Time{}
```
Comment (dborovcanin) [inline auth/jwt/tokenizer.go]: Avoid using `else`, something like:
```go
	key.ExpiresAt = time.Time{}
	if c.ExpiresAt != nil && !c.ExpiresAt.IsZero() {
		key.ExpiresAt = c.ExpiresAt.Time.UTC()
	}
```

File: auth/postgres/groups.go [hunk 36]
```diff
@@ -479,7 +483,11 @@ func (gr groupRepository) Unassign(ctx context.Context, groupID string, ids ...s
 		}
 
 		if _, err := tx.NamedExecContext(ctx, qDel, dbg); err != nil {
-			tx.Rollback()
+			if rollbackErr := tx.Rollback(); rollbackErr != nil {
+				err = errors.Wrap(err, rollbackErr)
```
Comment (dborovcanin) [inline auth/postgres/groups.go]: Reverse: `err = errors.Wrap(rollbackErr, err)`.

File: auth/service.go [hunk 37]
```diff
@@ -108,6 +108,7 @@ func (svc service) Issue(ctx context.Context, token string, key Key) (Key, strin
 		return Key{}, "", ErrInvalidKeyIssuedAt
 	}
 	switch key.Type {
+
```
Comment (dborovcanin) [inline auth/service.go]: Remove an empty line.

File: auth/service_test.go (hunk 34 above)
Comment (dborovcanin) [reply auth/service_test.go:1086]: @AryanGodara In that case, extract it as a package-level constant en reuse all over tests.

File: bootstrap/redis/producer/streams.go [hunk 38]
```diff
@@ -129,10 +123,10 @@ func (es eventStore) Bootstrap(ctx context.Context, externalKey, externalID stri
 
 	if err != nil {
 		ev.success = false
+		return cfg, err
```
Comment (dborovcanin) [inline bootstrap/redis/producer/streams.go]: No need to prevent publishing with the error indicator so we emit all events, even those unsuccessful.

File: cmd/certs/main.go (hunk 27 above)
Comment (dborovcanin) [reply cmd/certs/main.go]: Can you, please, send me the exact `golangci-lint` command to test this? The command we have in CI script (`golangci-lint run --no-config --disable-all --enable gosimple --enable errcheck --enable govet --enable unused --enable goconst --timeout 3m`) does not show me any problems with original code.

File: readers/cassandra/messages.go [hunk 39]
```diff
@@ -128,7 +128,9 @@ func buildQuery(chanID string, rpm readers.PageMetadata) (string, []interface{})
 	if err != nil {
 		return condCQL, vals
 	}
-	json.Unmarshal(meta, &query)
+	if err = json.Unmarshal(meta, &query); err != nil {
```
Comment (dborovcanin) [inline readers/cassandra/messages.go]: Don't shadow the original `err`: `if err := json.Unmarshal(meta, &query); err != nil {`.
Also, please check the later usage (L43 of this file) and if potential error should be handled.

File: readers/mocks/messages.go [hunk 40]
```diff
@@ -39,8 +39,13 @@ func (repo *messageRepositoryMock) ReadAll(chanID string, rpm readers.PageMetada
 	}
 
 	var query map[string]interface{}
-	meta, _ := json.Marshal(rpm)
-	json.Unmarshal(meta, &query)
+	meta, err := json.Marshal(rpm)
+	if err != nil {
+		return readers.MessagesPage{}, err
+	}
+	if err = json.Unmarshal(meta, &query); err != nil {
```
Comment (dborovcanin) [inline readers/mocks/messages.go]: The same here: `if err := json.Unmarshal(meta, &query); err != nil {`.

File: readers/timescale/messages.go [hunk 41]
```diff
@@ -121,7 +121,9 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) string {
 	if err != nil {
 		return condition
 	}
-	json.Unmarshal(meta, &query)
+	if err = json.Unmarshal(meta, &query); err != nil {
```
Comment (dborovcanin) [inline readers/timescale/messages.go]: And here.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: cmd/certs/main.go (hunk 27 above)
Comment (AryanGodara) [reply cmd/certs/main.go]: This is no longer highlighted after updating the linter, so I've reverted back to original.
Maybe they fixed the linter in the update.
But when I use the original
```
        block, _ := pem.Decode(b)
	if block == nil {
		logger.Fatal("No PEM data found, failed to decode CA")
	}

	caCert, err = x509.ParseCertificate(block.Bytes)    <--- IDE warns here about possible nil pointer dereference even after checking for nil above
	if err != nil {
		return tlsCert, caCert, errors.Wrap(errFailedCertDecode, err)
	}
```

Comment (AryanGodara) [reply cmd/certs/main.go]: @dborovcanin Should I leave this as it was(original code) or change to switch statement?

File: bootstrap/redis/producer/streams.go (hunk 38 above)
Comment (AryanGodara) [reply bootstrap/redis/producer/streams.go]: @dborovcanin I don't understand what this means 😅 . Should I remove the return statement, and let es.Add() run? I don't understand the reasoning

Comment (dborovcanin) [reply bootstrap/redis/producer/streams.go]: Yes.

File: scripts/ci.sh [hunk 42]
```diff
@@ -85,7 +86,7 @@ setup() {
 
 run_test() {
     echo "Running lint..."
-    golangci-lint run --no-config --disable-all --enable gosimple --enable govet --enable unused --enable deadcode --timeout 3m
+    golangci-lint run --no-config --disable-all --enable gosimple --enable errcheck --enable govet --enable unused --enable goconst --timeout 3m
```
Comment (rodneyosodo) [inline scripts/ci.sh:89]:
```suggestion
    golangci-lint run --no-config --disable-all --enable gosimple --enable gofmt --enable errcheck --enable errorlint --enable govet --enable unused --enable goconst --enable bodyclose  --enable misspell --enable nilerr --enable revive --enable unconvert --timeout 3m
```

File: cmd/certs/main.go (hunk 27 above)
Comment (dborovcanin) [reply cmd/certs/main.go]: Let's keep it as it is.

File: scripts/ci.sh (hunk 42 above)
Comment (AryanGodara) [reply scripts/ci.sh:89]: revive seems to give a lot of errors (51 issues found, and some of them are not useful, but will rather make the code more confusing.
Eg.: var varName string = "" ; or var varName bool = false;   <---Remove the = "" or = false, since it's the default value. This makes the code more confusing imo

Spellcheck is useful to run locally before commiting. But it also flags mqtt mosquitto, so we may have to remove it from the script. But it was very useful

Comment (dborovcanin) [reply scripts/ci.sh:89]: If that's the case, let's not address those more complex linters in this PR, but make a separate ticket for them. Those will require probably require config files, and maybe even integration with IDE. @AryanGodara Please do not forget to open a ticket for this and once you do, feel free to resolve this conversation and mark the PR as ready.

Comment (AryanGodara) [reply scripts/ci.sh:89]: #1774 new issue opened here. Resolving this conversation

**Review by drasko — APPROVED** [review]
LGTM
